packages/forms-web-app/src/lib/multifile-upload.js

```diff
@@ -0,0 +1,44 @@
+const getValidFiles = (errors, files) => {
+	// This function is called after the req has passed through validation middleware.
+	// There can be valid and invalid files in a multi-file upload, and the valid files need
+	// uploading, whilst the invalid ones do not. We will determine the valid files from the
+	// validation `errors` object. During testing it was found `md5` is sometimes not unique(!)
+	// though `tempFilePath` does appear to always be unique due to its use of timestamps.
+
+	const erroredFilesByTempFilePath = Object.values(errors).reduce((acc, error) => {
+		if (!error.value || !error.value.tempFilePath) {
+			return acc;
+		}
+		return [...acc, error.value.tempFilePath];
+	}, []);
+	return files.filter((file) => erroredFilesByTempFilePath.includes(file.tempFilePath) === false);
+};
+
+const removeFiles = (files, removedFiles) => {
+	for (const removedFile of removedFiles) {
+		files = files.filter((file) => file.name !== removedFile.name);
+	}
+	return files;
+};
+
+const uploadedFileData = (file, document) => {
+	return {
+		id: document.id,
+		name: file.name,
+		// needed for MoJ multi-file upload display
+		message: {
+			text: file.name
+		},
+		fileName: file.name,
+		originalFileName: file.name,
+		// needed for Cypress testing
```

Comment:
It would be good to dig into this need a bit more, ideally we wouldn't be changing the production code to make it easier to test, although I understand that sometimes this is needed so just wanted to make sure we had explored all alternatives to this.

Reply:
good call - I will investigate today - this object (and comment) is a cut and paste extraction from existing code so I don't know the history: we probably don't need to keep this in, especially as we're not using the cypress e2e tests.

Reply:
I think you are right that this needs digging into more - how many of the fields that we populate with data from the document API response (after successfully uploading a file) do we actually need? There is also a comment in the `document.js` controller along the lines of 'should we be returning any document data beyond the document's ID upon successful upload?' - this data is available via the `serveDocumentById()` function, accessible via a get request. I would suggest adding this to the tech debt ticket for further investigation.

Reply:
I also found an existing mapper in `src/mappers/document-mapper`, so have adapted that to include the required field for multi-file uploads and removed the mapping function from this module.